src/hb-coretext-font.cc

```diff
@@ -0,0 +1,355 @@
+  }
+  return false;
+}
+
+static unsigned int
+hb_coretext_get_nominal_glyphs (hb_font_t *font HB_UNUSED,
+				void *font_data,
+				unsigned int count,
+				const hb_codepoint_t *first_unicode,
+				unsigned int unicode_stride,
+				hb_codepoint_t *first_glyph,
+				unsigned int glyph_stride,
+				void *user_data HB_UNUSED)
+{
+  CTFontRef ct_font = (CTFontRef) font_data;
+
+  UniChar ch[MAX_GLYPHS];
+  CGGlyph cg_glyph[MAX_GLYPHS];
+  for (unsigned i = 0; i < count; i += MAX_GLYPHS)
+  {
+    unsigned c = (unsigned) hb_min ((int) MAX_GLYPHS, (int) count - (int) i);
+    for (unsigned j = 0; j < c; j++)
+    {
+      ch[j] = *first_unicode;
+      first_unicode = &StructAtOffset<const hb_codepoint_t> (first_unicode, unicode_stride);
+    }
+    CTFontGetGlyphsForCharacters (ct_font, ch, cg_glyph, c);
+    for (unsigned j = 0; j < c; j++)
+    {
+      *first_glyph = cg_glyph[j];
+      first_glyph = &StructAtOffset<hb_codepoint_t> (first_glyph, glyph_stride);
+    }
+  }
+
+  return count;
+}
+
+static hb_bool_t
+hb_coretext_get_variation_glyph (hb_font_t *font HB_UNUSED,
+				 void *font_data,
+				 hb_codepoint_t unicode,
+				 hb_codepoint_t variation_selector,
+				 hb_codepoint_t *glyph,
+				 void *user_data HB_UNUSED)
+{
+  // TODO. How?
+  return false;
+}
+
+static void
+hb_coretext_get_glyph_h_advances (hb_font_t* font, void* font_data,
+				  unsigned count,
+				  const hb_codepoint_t *first_glyph,
+				  unsigned glyph_stride,
+				  hb_position_t *first_advance,
+				  unsigned advance_stride,
+				  void *user_data HB_UNUSED)
+{
+  CTFontRef ct_font = (CTFontRef) font_data;
+
+  CGFloat ct_font_size = CTFontGetSize (ct_font);
+  CGFloat x_mult = (CGFloat) font->x_scale / ct_font_size;
+
+  CGGlyph cg_glyph[MAX_GLYPHS];
+  CGSize advances[MAX_GLYPHS];
+  for (unsigned i = 0; i < count; i += MAX_GLYPHS)
+  {
+    unsigned c = (unsigned) hb_min ((int) MAX_GLYPHS, (int) count - (int) i);
+    for (unsigned j = 0; j < c; j++)
+    {
+      cg_glyph[j] = *first_glyph;
+      first_glyph = &StructAtOffset<const hb_codepoint_t> (first_glyph, glyph_stride);
+    }
+    CTFontGetAdvancesForGlyphs (ct_font, kCTFontOrientationHorizontal, cg_glyph, advances, c);
+    for (unsigned j = 0; j < c; j++)
+    {
+      *first_advance = round (advances[j].width * x_mult);
+      first_advance = &StructAtOffset<hb_position_t> (first_advance, advance_stride);
+    }
+  }
+}
+
+
+static hb_bool_t
+hb_coretext_get_glyph_extents (hb_font_t *font,
+			       void *font_data,
+			       hb_codepoint_t glyph,
+			       hb_glyph_extents_t *extents,
+			       void *user_data HB_UNUSED)
+{
+  CTFontRef ct_font = (CTFontRef) font_data;
+
+  CGFloat ct_font_size = CTFontGetSize (ct_font);
+  CGFloat x_mult = (CGFloat) font->x_scale / ct_font_size;
+  CGFloat y_mult = (CGFloat) font->y_scale / ct_font_size;
+
+  CGGlyph glyphs[1] = { glyph };
+  CGRect bounds = ::CTFontGetBoundingRectsForGlyphs(ct_font,
+						    kCTFontDefaultOrientation, glyphs, NULL, 1);
```

Comment:
This causes a compiler warning:
```
../src/hb-coretext-font.cc:160:11: warning: 'kCTFontDefaultOrientation' is deprecated: first deprecated in macOS 10.11 - Deprecated [-Wdeprecated-declarations]
```
https://developer.apple.com/documentation/coretext/ctfontorientation/kctfontdefaultorientation?language=objc Suggests using `kCTFontOrientationDefault`.

If we use the new name someone is going to complain that we broke compilation on some ancient macOS version, so may be something like:
```diff
diff --git a/src/hb-coretext-font.cc b/src/hb-coretext-font.cc
index e06049d44..4e4a7dddf 100644
--- a/src/hb-coretext-font.cc
+++ b/src/hb-coretext-font.cc
@@ -142,6 +142,10 @@ hb_coretext_get_glyph_h_advances (hb_font_t* font, void* font_data,
 }


+#if MAC_OS_X_VERSION_MIN_REQUIRED < 101100
+# define kCTFontOrientationDefault kCTFontDefaultOrientation
+#endif
+
 static hb_bool_t
 hb_coretext_get_glyph_extents (hb_font_t *font,
                               void *font_data,
@@ -157,7 +161,7 @@ hb_coretext_get_glyph_extents (hb_font_t *font,

   CGGlyph glyphs[1] = { glyph };
   CGRect bounds = ::CTFontGetBoundingRectsForGlyphs(ct_font,
-                                                   kCTFontDefaultOrientation, glyphs, NULL, 1);
+                                                   kCTFontOrientationDefault, glyphs, NULL, 1);

   extents->x_bearing = round (bounds.origin.x * x_mult);
   extents->y_bearing = round (bounds.origin.y * y_mult);
```